examples_tests/33.Draw3DLine/main.cpp

```diff
@@ -0,0 +1,94 @@
+#define _IRR_STATIC_LIB_
+#include <irrlicht.h>
+
+#include "../../ext/Draw/CDraw3DLine.h"
+#include "../source/Irrlicht/COpenGLExtensionHandler.h"
+
+using namespace irr;
+using namespace core;
+
+int main()
+{
+    irr::SIrrlichtCreationParameters params;
+    params.Bits = 24;
+    params.ZBufferBits = 24;
+    params.DriverType = video::EDT_OPENGL;
+    params.WindowSize = core::dimension2d<uint32_t>(1280, 720);
+    params.Fullscreen = false;
+    params.Vsync = true;
+    params.Doublebuffer = true;
+    params.Stencilbuffer = false;
+    params.AuxGLContexts = 16;
+    IrrlichtDevice* device = createDeviceEx(params);
+
+    if (device == 0)
+        return 1;
+
+    video::IVideoDriver* driver = device->getVideoDriver();
+    scene::ISceneManager* smgr = device->getSceneManager();
+    auto draw3DLine = ext::draw::CDraw3DLine::create(driver);
+
+    auto camera = smgr->addCameraSceneNodeFPS(0,100.0f,0.001f);
+
+    camera->setPosition(core::vector3df(-4,0,0));
+    camera->setTarget(core::vector3df(0,0,0));
+    camera->setNearValue(0.01f);
+    camera->setFarValue(10.0f);
+
+    smgr->setActiveCamera(camera);
+
+    uint64_t lastFPSTime = 0;
+
+    while(device->run())
+    if (device->isWindowActive())
+    {
+        driver->beginScene(true, false, video::SColor(255,255,255,255));
+
```

Comment:
if there is no `smgr->drawAll` or at least `smgr->OnAnimate` then the active camera will not be updated, and hence you will get no change in the view-proj matrix and ergo your line won't move

Reply:
added smgr->drawAll()